astropy/nddata/tests/test_nduncertainty.py

```diff
@@ -354,3 +364,36 @@ def test_assigning_uncertainty_with_bad_unit_to_parent_fails(NDClass,
     v = UncertClass([1, 1], unit=u.second)
     with pytest.raises(u.UnitConversionError):
         ndd.uncertainty = v
+
+
+@pytest.mark.parametrize(
+    'UncertClass', uncertainty_types_with_conversion_support
+)
+def test_self_conversion_via_variance_supported(UncertClass):
+    uncert = np.arange(1, 11).reshape(2, 5) * u.adu
+    start_uncert = UncertClass(uncert)
+    final_uncert = start_uncert.represent_as(UncertClass)
+    assert_array_equal(start_uncert.array, final_uncert.array)
+    assert start_uncert.unit == final_uncert.unit
+
+
+@pytest.mark.parametrize(
+    'UncertClass', uncertainty_types_with_conversion_support
+)
+def test_conversion_to_from_variance_supported(UncertClass):
```

Comment:
What I had in mind here was checking that if you start with one class and convert to variance, then the variance values are correct. Something like this:

```
@pytest.mark.parametrize(
    'UncertClass', uncertainty_types_with_conversion_support,
    'to_variance_func', (lambda x: x ** 2, lambda x: x, lambda x: 1 / x)
)
def test_conversion_to_from_variance_supported(UncertClass, to_variance_func):
    uncert = np.arange(1, 11).reshape(2, 5) * u.adu
    start_uncert = UncertClass(uncert)
    var_uncert = start_uncert.represent_as(VarianceUncertainty)
    assert_allclose(to_variance_func(start_uncert.array), var_uncert.array)
    # unit check too
```

I think what you have added does check the roundtripping but the intermediate result is still unchecked (I think).

Reply:
Ah, ok, I'll add that as well.